rust/headless-client/src/linux.rs

```diff
@@ -10,31 +10,62 @@ use connlib_shared::{
 use firezone_cli_utils::setup_global_subscriber;
 use futures::{SinkExt, StreamExt};
 use secrecy::SecretString;
-use std::{
-    future,
-    net::IpAddr,
-    path::{Path, PathBuf},
-    str::FromStr,
-    task::Poll,
-};
+use std::{future, net::IpAddr, path::PathBuf, str::FromStr, task::Poll};
 use tokio::{
     net::{UnixListener, UnixStream},
     signal::unix::SignalKind,
 };
 use tokio_util::codec::LengthDelimitedCodec;
 
+const SOCK_PATH: &str = "/run/firezone-client.sock";
+
 pub async fn run() -> Result<()> {
     let cli = Cli::parse();
     let (layer, _handle) = cli.log_dir.as_deref().map(file_logger::layer).unzip();
     setup_global_subscriber(layer);
 
     match cli.command() {
+        Cmd::Auto => {
+            if let Some(token) = token(&cli).await? {
+                run_standalone(cli, &token).await
+            } else {
+                run_daemon(cli).await
+            }
+        }
         Cmd::Daemon => run_daemon(cli).await,
-        Cmd::Standalone => run_standalone(cli).await,
+        Cmd::DebugIpcClient => run_debug_ipc_client(cli).await,
+        Cmd::Standalone => {
+            let token = token(&cli)
+                .await?
+                .context("Need a token to run as standalone Client")?;
+            run_standalone(cli, &token).await
+        }
+    }
+}
+
+async fn token(cli: &Cli) -> Result<Option<SecretString>> {
+    if let Some(token) = &cli.token {
+        // Token was provided in CLI args or env var
+        // Not very secure, but we do get the token
+        return Ok(Some(token.clone().into()));
     }
+
+    let path = PathBuf::from("/etc")
+        .join(connlib_shared::BUNDLE_ID)
+        .join("token.txt");
+    let Ok(bytes) = tokio::fs::read(path).await else {
+        // Can't read the file for some reason, so there's no token on disk
+        return Ok(None);
+    };
+    let s = String::from_utf8(bytes)?;
+    let token = s.trim().to_string();
+
+    // Token loaded from disk
+    Ok(Some(token.into()))
 }
 
-async fn run_standalone(cli: Cli) -> Result<()> {
+async fn run_standalone(cli: Cli, token: &SecretString) -> Result<()> {
+    tracing::info!("run_standalone");
```

Comment:
```suggestion
    tracing::info!("Running in standalone mode");
```

Slight preference for a "proper" message.

Reply:
Sure I guess, since power users may see it in the logs. If it was debug I'd still prefer `run_standalone` since it matches the fn name in grep.